### ISSUE TITLE
DX: xdebug does not halt at breakpoints

### ISSUE THREAD

**Issue by staabm** [rectorphp/rector#8182]
# Bug Report

when I try to debug why a rule is not applied in a certain project I run e.g.

```
 vendor/bin/rector process app/homie/lib/buyer/service/InviteService.php -a app/homie/phpstan-bootstrap.php  -vvv --debug --xdebug
```

xdebug does not halt the process because rector is running in parallel mode and workers don't know about xdebug, as the xdebug flag is not known to them.

I think for improved DX we either should make workers aware of the `--xdebug` flag, so xdebug can halt or we should alternatively automatically disable parallel when `--debug --xdebug` is used, as the user tries to debug while using a debugger.

**Comment by TomasVotruba** [comment]
> I think for improved DX we either should make workers aware of the --xdebug flag, 

This is a way to go. That's how it should work :+1:

**Comment by staabm** [comment]
looking thru the worker command and doing some step debugging it seems the `--xdebug` is passed thru to workers, when doing a full project scan.

I was not yet able to verify how it works when processing just a single file, as for some reason xdebug does not work in that case in my testing

**Comment by samsonasik** [comment]
@staabm this probably related with proc_open() downgrade bug on `XdebugHandler`:

- https://github.com/rectorphp/rector-downgrade-php/issues/198

that fixed at :

- https://github.com/rectorphp/rector-downgrade-php/pull/199

could you try latest `dev-main`, if this still an issue, that probably different issue than this.

**Comment by TomasVotruba** [comment]
Closing as duplicate of https://github.com/rectorphp/rector/issues/8060,
there is also proposed solution that might help to fix it :pray:
